Services/CO.CDP.EntityVerification/Ppon/OrganisationUpdatedSubscriber.cs

```diff
@@ -0,0 +1,38 @@
+using CO.CDP.EntityVerification.Events;
+using CO.CDP.EntityVerification.Persistence;
+using CO.CDP.MQ;
+
+namespace CO.CDP.EntityVerification.Ppon;
+
+public class OrganisationUpdatedSubscriber(
+    IPponRepository pponRepository)
+    : ISubscriber<OrganisationUpdated>
+{
+    public class OrganisationUpdatedException(string message, Exception? cause = null) : Exception(message, cause)
+    {
+        public class NotFoundPponException(string message, Exception? cause = null)
+            : Exception(message, cause);
+    }
+
+    public async Task Handle(OrganisationUpdated @event)
+    {
+        var pponIdentifier = @event.FindIdentifierByScheme(Events.Identifier.PponSchemeName);
+
+        if (pponIdentifier != null)
+        {
+            var pponToUpdate = await pponRepository.FindPponByPponIdAsync(pponIdentifier.Id);
+
+            if (pponToUpdate != null)
+            {
+                var identifiersToPersist = Persistence.Identifier.GetPersistenceIdentifiers(@event.AllIdentifiers());
+
+                foreach (var identifier in identifiersToPersist)
+                {
+                    pponToUpdate.Identifiers.Add(identifier);
+                }
```

Comment:
Since the `OrganisationUpdated` event always contains all the latest organisation details, the subscriber is responsible for reconciling changes. I know it's not the usual case for events, but that's what we went for by deciding to publish coarse events with most organisation details. If we don't check which identifiers were already here, we will be appending them every time.

Reply:
Now only adds Identifiers that do not exist.